cmp/ag-grid/AgGridModel.js

```diff
@@ -257,33 +258,61 @@ export class AgGridModel {
     setSortState(sortState) {
         this.throwIfNotReady();
 
-        const sortModel = cloneDeep(sortState),
-            {agApi, agColumnApi} = this,
-            isPivot = agColumnApi.isPivotMode(),
-            havePivotCols = !isEmpty(agColumnApi.getPivotColumns());
-
-        sortModel.forEach(sort => {
-            if (isArray(sort.colId)) {
-                if (isPivot && havePivotCols) {
-                    // Find the appropriate secondary column
-                    const secondaryCols = agColumnApi.getSecondaryColumns(),
-                        col = secondaryCols.find(
-                            it => isEqual(sort.colId, this.getPivotColumnId(it)));
-
-                    if (col) {
-                        sort.colId = col.colId;
-                    } else {
-                        console.warn(
-                            'Could not find a secondary column to associate with the pivot column path',
-                            sort.colId);
-                    }
+        const sortedColumnState = cloneDeep(sortState),
+            [primaryColumnState, secondaryColumnState] = partition(sortedColumnState, it => !isArray(it.colId)),
+            {agColumnApi: colApi, agApi} = this,
+            isPivot = colApi.isPivotMode(),
+            havePivotCols = !isEmpty(colApi.getPivotColumns());
+            
+        // ag-Grid does not allow "secondary" columns to be manipulated by applyColumnState
+        // so this approach is required for setting sort config on secondary columns.
+        if (isPivot && havePivotCols && !isEmpty(secondaryColumnState)) {
+            // 1st clear all pre-exisiting primary column sorts
+            // with an explicit clear of the auto_group column, 
+            // which is not cleared by the defaultState config.
+            colApi.applyColumnState({
+                state: [{
+                    colId: AgGridModel.AUTO_GROUP_COL_ID,
+                    sort: null,
+                    sortIndex: null
+                }],
+                defaultState: {
+                    sort: null,
+                    sortIndex: null
+                }
+            });
+
+            // 2nd clear all pre-exisiting secondary column sorts
+            colApi.getSecondaryColumns().forEach(col => {
+                if (col) {
+                    col.setSort(undefined);
```

Comment:
Why undefined here instead of null?

Reply:
When using `applyColumnState`, `undefined` means do nothing, `null` means set to none, not cleared.
But when using the setSort & setSortIndex methods directly, to clear all sort settings as if no sort
had ever been specified, `undefined` must be used.

see https://github.com/ag-grid/ag-grid/blob/3ed885a9cde6742d5c2469093f02b424d9c2bb05/community-modules/core/src/ts/sortController.ts#L103
see https://github.com/ag-grid/ag-grid/blob/3ed885a9cde6742d5c2469093f02b424d9c2bb05/community-modules/core/src/ts/sortController.ts#L66